xdsl/backend/riscv/targets.py

```diff
@@ -93,6 +99,17 @@ class ABISpec:
     "elf" is the default for most
     """
 
+    # stack alignment:
+    stack_alignment: int = 128
+    """
+    The stack grows downwards (towards lower addresses) and the stack pointer shall be
+    aligned to a 128-bit boundary upon procedure entry. The first argument passed on the
+    stack is located at offset zero of the stack pointer on function entry; following
+    arguments are stored at correspondingly higher addresses.
```

Comment:
lower addresses?

Reply:
I'm confused, I see that it's verbatim the text form the link but I'm not sure how it can grow downwards and for following arguments being at higher addresses...

Reply:
The caller "pushes" arguments to the stack (if they don't fit into registers), once control is given over to the calling function, the stack pointer is pointing to the *end* of the arguments, so that the empty stack lies "below" the stack pointer, and the arguments directly above.

to illustrate:

```
+--------------------------+
|  Calling functions stack |
|                          |
+--------------------------+
|    Padding if needed     |
+--------------------------+
|     stack argument n     |
|     stack argument n-1   |
|            ...           |
|     stack argument 0     |
+--------------------------+ <--- Stack pointer, aligned to 128 bits
|           empty          |
+--------------------------+
```

The fact that stack arguments are passed like this is likely irrelevant for us for now,
but we may need to consider that the `sp` should be 128 bit aligned when calling other
functions, as they likely rely on this.

Reply:
ah this makes sense, and explains why gcc and llvm allocate more stack space than they need in my explorations, thank you

Reply:
I'm fond of ASCII art in comments, can we add this in source pls?

Reply:
Will do!